common.go

```diff
@@ -607,6 +612,22 @@ type Config struct {
 	// session tickets, instead of SessionTicketKey.
 	SessionTicketSealer SessionTicketSealer
 
+	// AcceptDelegatedCredential is true if the client is willing to negotiate
+	// the delegated credential extension.
+	//
+	// This value has no meaning for the server.
+	//
+	// See https://tools.ietf.org/html/draft-ietf-tls-subcerts.
+	AcceptDelegatedCredential bool
+
+	// GetDelegatedCredential returns a DelegatedCredential for use with the
+	// delegated credential extension based on the ClientHello and TLS version
+	// selected for the session. If this is nil, then the server will not offer
+	// a DelegatedCredential.
+	//
+	// This value has no meaning for the client.
+	GetDelegatedCredential func(*ClientHelloInfo, uint16) (*DelegatedCredential, error)
```

Comment:
This is necessary because the DC is bound to the TLS version in which it's to be used. `ClientHelloInfo` provides the range of version supported by the client, but doesn't reflect the server's decision. Passing in the version here allows us to choose the DC after the server has picked the DC.

Reply:
Hm. Do you know the purpose of binding to the version?

I think there was a GREASE-releated project where every few days, Chrome and Cloudflare's edge start speaking a new version of TLS. This puts a wrench in that.

Reply:
But GREASE doesn't change the TLS version, does it? It just offers random extensions and ciphersuites.

Here's what the spec says. Section 2.1: "Delegated credentials allow signed messages to be bound to specific versions of TLS. This prevents them from being used for other protocols if a service owner allows multiple versions of TLS." Section 4: "It is bound to the protocol version that is negotiated. This is intended to avoid cross-protocol attacks with signing oracles."